.github/workflows/gradle-enterprise-conventions-maven-extension-verification.yml

```diff
@@ -0,0 +1,63 @@
+name: Verify Maven Convention Extension
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build Gradle Enterprise Conventions Maven Extension
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+          cache: maven
+      - name: Build Gradle Enterprise Conventions Maven Extension
+        run: ./mvnw clean install -Dgradle.enterprise.url=https://ge.solutions-team.gradle.com
+        working-directory: gradle-enterprise-conventions-maven-extension/extension
+        env:
+          GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_SOLUTIONS_ACCESS_TOKEN }}
+      - name: Upload extension
+        uses: actions/upload-artifact@v3
+        with:
+          name: gradle-enterprise-conventions-maven-extension
+          path: ~/.m2/repository/com/example/gradle-enterprise-conventions-maven-extension
+  verification:
+    name: Verify Example Build for Maven ${{ matrix.maven-version }}
+    needs: [build]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        maven-version: ['3.6.3', '3.8.8', '3.9.1']
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+          cache: maven
+      - name: Download Extension
+        uses: actions/download-artifact@v3
+        with:
+          name: gradle-enterprise-conventions-maven-extension
+          path: ~/.m2/repository/com/example/gradle-enterprise-conventions-maven-extension
+      - name: Set Maven Version
+        run: ./mvnw wrapper:wrapper -Dmaven=${{ matrix.maven-version }} -Dgradle.enterprise.url=https://ge.solutions-team.gradle.com
+        working-directory: gradle-enterprise-conventions-maven-extension/example-build
+        env:
+          GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_SOLUTIONS_ACCESS_TOKEN }}
+      - name: Verify Example Build
+        run: ./mvnw clean verify -Dgradle.enterprise.url=https://ge.solutions-team.gradle.com
```

Comment:
Is the intention here to check that the convention extension has been applied to the build?

Reply:
It's a rudimentary check just to make sure that the build succeeds with the convention extension. It won't actually verify the extension has been applied I suppose.

Reply:
It could be a nice improvement.
I can think of a custom extension creating a dummy file for which presence would be checked to assess E2E validation.

Reply:
That is a good idea. I will look into adding something like that

Reply:
For Gradle I added a check that a build scan was published. The `setup-gradle` action makes this very easy. https://github.com/gradle/develocity-build-config-samples/blob/4e678086c664fcefc8d7be9913d7f05d84cee764/.github/workflows/convention-develocity-gradle-plugin-verification.yml#L62-L64

It would be nice if `maven-setup` could do the same. https://github.com/gradle/develocity-actions/tree/main/maven-setup